pkg/flink/internal/controller/interactive_output_controller.go

```diff
@@ -22,14 +23,16 @@ type InteractiveOutputController struct {
 	tableView     components.TableViewInterface
 	resultFetcher types.ResultFetcherInterface
 	isRowViewOpen bool
+	outputFormat  func() config.OutputFormat
 	debug         bool
 }
 
-func NewInteractiveOutputController(tableView components.TableViewInterface, resultFetcher types.ResultFetcherInterface, debug bool) types.OutputControllerInterface {
+func NewInteractiveOutputController(tableView components.TableViewInterface, resultFetcher types.ResultFetcherInterface, outputFormat func() config.OutputFormat, debug bool) types.OutputControllerInterface {
```

Comment:
I don't really like that we have to pass functions around `outputFormat func() config.OutputFormat`. I think we should instead extract the UserProperties out of the store and then pass those directly. Wdyt?
You don't have to do it now if it's to big of a refactor though

Reply:
Some refactoring was involved, but done

Reply:
very cool, thanks!